### PR TITLE
Map Vim window rotation types to actions

### PR DESCRIPTION
**Pull request by KevinSjoberg**
Depends on the work done in https://github.com/onivim/reason-libvim/pull/63. This pull request add the ability to trigger window rotation using `c-w r` and `c-w R`. 🎉

### CODE REVIEW

File: src/editor/Store/VimStoreConnector.re [hunk 1]
```diff
@@ -154,28 +154,35 @@ let start = (getState: unit => Model.State.t, getClipboardText) => {
...
-        Model.WindowTree.getEditorGroupIdFromSplitId(
-          v,
-          currentState.windowManager.windowTree,
-        );
-      switch (editorId) {
-      | Some(ed) => dispatch(Model.Actions.WindowSetActive(v, ed))
-      | None => ()
+      switch (movementType) {
+      | FullLeft
+      | OneLeft => move(Model.WindowManager.moveLeft, currentState)
```
Comment (KevinSjoberg) [inline src/editor/Store/VimStoreConnector.re]: I was not entirely sure what's best practice here. I don't technically have to pass `currentState` to `move` due to lexical scope. `currentState` is defined in the outer scope. However, passing `currentState` makes the dependency obvious.

Thoughts?

Comment (tcoopman) [reply src/editor/Store/VimStoreConnector.re]: You are not using `state` but `currentState` in your move function, so now the argument makes no sense at all.
I probably wouldn't pass the state in, but not sure what the best practices are for this.

Comment (KevinSjoberg) [reply src/editor/Store/VimStoreConnector.re]: Oh, your're right. Nice catch! 🙂 Well, given that the `move` function do use the `state` parameter, does it make sense?

Comment (tcoopman) [reply src/editor/Store/VimStoreConnector.re]: I don't think it matters much, and think it's more of a style thing.

**Comment by bryphe** [comment]
Awesome! Just tested it out and it works great 👍 Thanks for the work on this @KevinSjoberg , and the help reviewing @tcoopman . 

Change looks good to me - just needs an `esy format` (and might need to be merged w/ `master` to pick up the latest `reason-libvim`) - and should be green after that.

**Comment by KevinSjoberg** [comment]
@bryphe rebased on latest master and fixed formatting.

**Comment by bryphe** [comment]
@KevinSjoberg - awesome! Bringing this in now - thank you!

**Pull request merged**